actors/builtin/miner/miner_state.go

```diff
@@ -875,68 +838,42 @@ func (st *State) UnlockUnvestedFunds(store adt.Store, currEpoch abi.ChainEpoch,
 // Unlocks all vesting funds that have vested before the provided epoch.
 // Returns the amount unlocked.
 func (st *State) UnlockVestedFunds(store adt.Store, currEpoch abi.ChainEpoch) (abi.TokenAmount, error) {
-	vestingFunds, err := adt.AsArray(store, st.VestingFunds)
+	vestingFunds, err := st.LoadVestingFunds(store)
 	if err != nil {
-		return big.Zero(), err
-	}
-
-	amountUnlocked := abi.NewTokenAmount(0)
-	lockedEntry := abi.NewTokenAmount(0)
-	var toDelete []uint64
-	var finished = fmt.Errorf("finished")
-
-	// Iterate vestingFunds  in order of release.
-	err = vestingFunds.ForEach(&lockedEntry, func(k int64) error {
-		if k < int64(currEpoch) {
-			amountUnlocked = big.Add(amountUnlocked, lockedEntry)
-			toDelete = append(toDelete, uint64(k))
-		} else {
-			return finished // stop iterating
-		}
-		return nil
-	})
-
-	if err != nil && err != finished {
-		return big.Zero(), err
-	}
-
-	err = vestingFunds.BatchDelete(toDelete)
-	if err != nil {
-		return big.Zero(), errors.Wrapf(err, "failed to delete locked fund during vest: %v", err)
+		return big.Zero(), xerrors.Errorf("failed to load vesting funds: %w", err)
 	}
 
+	amountUnlocked := vestingFunds.unlockVestedFunds(currEpoch)
 	st.LockedFunds = big.Sub(st.LockedFunds, amountUnlocked)
 	Assert(st.LockedFunds.GreaterThanEqual(big.Zero()))
-	st.VestingFunds, err = vestingFunds.Root()
+
+	err = st.SaveVestingFunds(store, vestingFunds)
 	if err != nil {
-		return big.Zero(), err
+		return big.Zero(), xerrors.Errorf("failed to save vesing funds: %w", err)
 	}
 
 	return amountUnlocked, nil
 }
 
 // CheckVestedFunds returns the amount of vested funds that have vested before the provided epoch.
 func (st *State) CheckVestedFunds(store adt.Store, currEpoch abi.ChainEpoch) (abi.TokenAmount, error) {
-	vestingFunds, err := adt.AsArray(store, st.VestingFunds)
+	vestingFunds, err := st.LoadVestingFunds(store)
 	if err != nil {
-		return big.Zero(), err
+		return big.Zero(), xerrors.Errorf("failed to load vesting funds: %w", err)
 	}
 
 	amountUnlocked := abi.NewTokenAmount(0)
-	lockedEntry := abi.NewTokenAmount(0)
-	var finished = fmt.Errorf("finished")
-
-	// Iterate vestingFunds  in order of release.
-	err = vestingFunds.ForEach(&lockedEntry, func(k int64) error {
-		if k < int64(currEpoch) {
-			amountUnlocked = big.Add(amountUnlocked, lockedEntry)
-		} else {
-			return finished // stop iterating
+
+	for i := range vestingFunds.Funds {
+		vf := vestingFunds.Funds[i]
+		epoch := vf.Epoch
+		amount := vf.Amount
+
+		if epoch >= currEpoch {
+			break
 		}
-		return nil
-	})
-	if err != nil && err != finished {
-		return big.Zero(), err
+
+		amountUnlocked = big.Add(amountUnlocked, amount)
```

Comment:
nit: "amount vested" because we're not actually unlocking.

Reply:
Good catch ! This is done.